community/kernel/src/test/java/org/neo4j/kernel/impl/store/kvstore/AbstractKeyValueStoreTest.java

```diff
@@ -80,6 +84,21 @@ public String toString()
     };
 
     @Test
+    @Resources.Life( STARTED )
+    public void retryLookupOnConcurrentStoreStateChange() throws IOException
+    {
+        Store testStore = resourceManager.managed( createTestStore( TimeUnit.DAYS.toMillis( 2 ) ) );
+        ProgressiveState progressiveState = mock( ProgressiveState.class );
+        CountingErroneousReader countingErroneousReader = new CountingErroneousReader( testStore, progressiveState );
+        testStore.put( "test", "value" );
+
+        testStore.lookup( "test", countingErroneousReader );
+
+        assertEquals( 1, countingErroneousReader.getInvocationCounter() );
```

Comment:
Should we not have the next state contain the key we are looking for (at least as a separate test), so that we can observe that the retry actually happens (by observing the invocation counter having a higher value despite the number of invocations to `testStore.lookup` only being 1).